File: Java Fundamentals-Jan2020/src/n_AssociativeArrays_Exercise/SoftUniParking.java
package n_AssociativeArrays_Exercise;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Scanner;

public class SoftUniParking {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine());

        Map<String, String> cars = new LinkedHashMap<>();

        for (int i = 0; i < n; i++) {
            String[] tokens = sc.nextLine().split("\\s+");

            String action = tokens[0];
            String name = tokens[1];

            switch (action) {
                case "register":
                    String plate = tokens[2];
                    if (cars.containsKey(name)) {
                        System.out.printf("ERROR: already registered with plate number %s%n", cars.get(name));
                    } else {
                        cars.put(name, plate);
                        System.out.printf("%s registered %s successfully%n", name, plate);
                    }
                    break;
                case "unregister":
                    if (!cars.containsKey(name)) {
                        System.out.printf("ERROR: user %s not found%n", name);
                    } else {
                        cars.remove(name);
                        System.out.printf("%s unregistered successfully%n", name);
                    }
                    break;
            }
        }
        //output
        cars.forEach((k, v) -> System.out.println(String.format("%s => %s",k, v)));
    }
}

File: Programming-for-QA-sep2023/15.DictionariesLambdaLINQ/08.SoftUniParking/SoftUniParking.cs
/*
 * The program receives 2 commands:
 *     - "register {username} {licensePlateNumber}" 
 *     - "unregister {username}"
 */

Dictionary<string, string> parking = new();

int n = int.Parse(Console.ReadLine());

for (int i = 0; i < n; i++)
{
    string[] input = Console.ReadLine().Split();

    string command = input[0];
    string username = input[1];

    if (command == "register")
    {
        string lpn = input[2];

        try
        {
            parking.Add(username, lpn);
            Console.WriteLine($"{username} registered {lpn} successfully");
        }
        catch (Exception)
        {
            Console.WriteLine($"ERROR: already registered with plate number {parking[username]}");
        }
    }
    else if (command == "unregister")
    {
        bool isUnregistered = parking.Remove(username);

        if (isUnregistered)
        {
            Console.WriteLine($"{username} unregistered successfully");
        }
        else
        {
            Console.WriteLine($"ERROR: user {username} not found");
        }
    }
}

// print all of the currently registered users and their license plates in the format: "{username} => {licensePlateNumber}"
parking.Select(x => $"{x.Key} => {x.Value}").ToList().ForEach(x => Console.WriteLine(x));
